### PR TITLE
Fixed a grammatical mistake: 'a HTTP ...' -> 'an HTTP ...'

### LINKED ISSUE #33728: A possible grammatical mistake in an error message
**Elasticsearch version** (`bin/elasticsearch --version`): 6.4.0, but relevant for most of the latest versions.

**Plugins installed**: _Irrelevant_ (none)

**JVM version** (`java -version`): _Irrelevant_

**OS version** (`uname -a` if on a Unix-like system): _Irrelevant_

**Description of the problem including expected versus actual behavior**:

When trying to access to port 9300 using http (using `curl elasticsearch_server:9300`, for example), an error message is return with the text **"This is not a HTTP port"**.
As far as I know and found, the text message should be **"This is not an HTTP port"**.

Moreover, "HTTP" is an acronym beginning with an H, [which is pronounced "aitch"](http://www.whichenglish.com/Better-English-Grammar/usage/haitch-or-aitch.html), and therefore should be preceded with "an" instead of "a".

Also, according to [this post](https://english.stackexchange.com/questions/629/when-should-i-use-a-versus-an-in-front-of-a-word-beginning-with-the-letter-h), any word beginning with a vowel should be preceded by "an", including words beginning with an 'h' when written but are pronounced without 'h'.

I intend to fix this myself very soon.

**Steps to reproduce**: _Irrelevant_

### PR DESCRIPTION
**Pull request by OrBin**
Fixed the mistake of 'a HTTP ...', as I have mentioned in #33728.

Closes #33728

### CODE REVIEW

File: docs/java-rest/low-level/configuration.asciidoc [hunk 1]
```diff
@@ -54,7 +54,7 @@ include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-config-basi
 --------------------------------------------------
 
 Preemptive Authentication can be disabled, which means that every request will be sent without
-authorization headers to see if it is accepted and, upon receiving a HTTP 401 response, it will
+authorization headers to see if it is accepted and, upon receiving HTTP 401 response, it will
```
Comment (nik9000) [inline docs/java-rest/low-level/configuration.asciidoc]: I think this should be "upon receiving *an* HTTP 401"

File: docs/reference/settings/security-settings.asciidoc [hunk 2]
```diff
@@ -78,7 +78,7 @@ The roles to associate with the anonymous user. Required.
 When `true`, an HTTP 403 response is returned if the anonymous user
 does not have the appropriate permissions for the requested action. The
 user is not prompted to provide credentials to access the requested
-resource. When set to `false`, a HTTP 401 is returned and the user
+resource. When set to `false`, HTTP 401 is returned and the user
```
Comment (nik9000) [inline docs/reference/settings/security-settings.asciidoc]: I think this reads a little better as "an HTTP 401 is returned"

Comment (OrBin) [reply docs/reference/settings/security-settings.asciidoc]: I'll take the wording from `docs/java-rest/low-level/configuration.asciidoc`: "an HTTP 401 **response** is returned".

**Comment by nik9000** [comment]
Looks good to me! Thanks @OrBin. For reference, if you see something like this some other time you dno't have to file an issue before opening a PR.

I'll merge this soon.

**Comment by nik9000** [comment]
Thanks again @OrBin!

**Pull request merged**